File: libsignal-service/src/test/java/org/whispersystems/signalservice/api/push/PushTransportDetailsTest.java
/*
 * Copyright 2023 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.push;

import org.junit.Test;
import org.whispersystems.signalservice.internal.push.PushTransportDetails;

import static org.junit.Assert.assertEquals;

public class PushTransportDetailsTest {

  @Test
  public void testV3Padding() {
    PushTransportDetails transportV3 = new PushTransportDetails();

    for (int i=0;i<159;i++) {
      byte[] message = new byte[i];
      assertEquals(159, transportV3.getPaddedMessageBody(message).length);
    }

    for (int i=159;i<319;i++) {
      byte[] message = new byte[i];
      assertEquals(319, transportV3.getPaddedMessageBody(message).length);
    }

    for (int i=319;i<479;i++) {
      byte[] message = new byte[i];
      assertEquals(479, transportV3.getPaddedMessageBody(message).length);
    }
  }
}

File: libsignal-service/src/test/java/org/whispersystems/signalservice/api/push/PushTransportDetailsTest.kt
/*
 * Copyright 2023 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.push

import assertk.assertThat
import assertk.assertions.hasSize
import org.junit.Test
import org.whispersystems.signalservice.internal.push.PushTransportDetails

class PushTransportDetailsTest {
  private val transportV3 = PushTransportDetails()

  @Test
  fun testV3Padding() {
    (0 until 79).forEach { i ->
      val message = ByteArray(i)
      assertThat(transportV3.getPaddedMessageBody(message)).hasSize(79)
    }

    (79 until 159).forEach { i ->
      val message = ByteArray(i)
      assertThat(transportV3.getPaddedMessageBody(message)).hasSize(159)
    }

    (159 until 239).forEach { i ->
      val message = ByteArray(i)
      assertThat(transportV3.getPaddedMessageBody(message)).hasSize(239)
    }
  }
}
